docs/dev_guide.md

```diff
@@ -1,31 +1,183 @@
 # Dev Guide
 
-## Adding a new param in GCSFuse
+## Contributing to GCSFuse
+
+**If you're interested in contributing a new feature, enhancement, or bug fix,
+please open a GitHub issue to discuss your proposal with the GCSFuse team first
+before making the contribution.** Make sure to describe the use case (what are
+you trying to solve and why?), and the proposed solution in the GitHub issue.
+
+**Note:** We make no guarantees that contribution requests are accepted, or will
+be rolled out as a Generally Available (GA) feature
+
+Rollout of any new feature should be done in 2 phases:
+
+**1. Experimental Phase:**
+
+* New features should initially be introduced behind an experimental flag. These
+  flags should be prefixed with `--experimental-` and marked as hidden in the
+  help output.
+* This signals to users that the feature is in development and may change or be
+  removed.
+* **Testing Requirements**:
+    * Thorough testing is a crucial requirement for any new features added to
+      GCSFuse even during the experimental phase. This helps ensure the
+      feature's stability, identify potential issues early, and prevent
+      regressions.
+    * Add unit tests for all new code written for the experimental feature. This
+      includes testing edge cases, boundary conditions, and error handling.
+    * Add composite tests to validate GCSFuse's end user functionality while
+      isolating the testing to the GCSFuse code itself. This allows for faster
+      and more controlled testing without relying on a real Google Cloud Storage
+      bucket.
+    * Add end-to-end tests for complete feature flows, simulating real-world use
+      with an actual Google Cloud Storage bucket.
+    * For more details on testing best practices, please refer to
+      the [testing guide](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/docs/dev_guide.md#testing-of-new-features)
+      below.
+      After the feature development is complete and the code is merged to the
+      mainline, the feature will roll out as an experimental feature in the next
+      GCSFuse release with the feature disabled by default.
+
+* After the feature development is complete and the code is merged to the
+  mainline, the feature will roll out as an experimental feature in the next
+  GCSFuse release with the feature disabled by default.
+
+* Developers should open a GitHub issue (
```

Comment:
Should the developers create a new github issue in addition to the one they started created before they started the experimental changes?

Reply:
Yes proposing it to be a new github issue. Original issue for proposal can be closed after experimental implementation is complete.